### DISCUSSION TITLE
how to add AddHttpActivities in elsa 3.1.0

### DISCUSSION THREAD

**Discussion by NiveditaGalwade** [elsa-workflows/elsa-core#5321] [Q&A]
similar to version services 2.13 mentioned on official elsa documention https://v2.elsaworkflows.io/docs/guides/document-approval

                .AddElsa(elsa => elsa
                    .UseEntityFrameworkPersistence(ef => ef.UseSqlite())
                    .AddConsoleActivities()
                    **.AddHttpActivities(elsaSection.GetSection("Server").Bind)**
                    .AddEmailActivities(elsaSection.GetSection("Smtp").Bind)
                    .AddQuartzTemporalActivities()
                    .AddWorkflowsFrom<Startup>()
                );
                
                how to add HttpActivities in elsa 3.1.0

**Comment by glime-ai** [comment]
In Elsa 3.1.0, you can add HTTP activities to your Elsa services configuration by using the `AddHttpActivities` extension method during the setup of your Elsa services in the `Program.cs` file. Here's an example of how you might do it:

```csharp
builder.Services.AddElsa(elsa => {
    // ... other configurations ...

    // Enable HTTP activities.
    elsa.AddHttpActivities();
    
    // ... other configurations ...
});
```

This will register the HTTP activities with the Elsa services, allowing you to use them in your workflows.

Please note that the exact syntax and method names can vary depending on the version of Elsa you are using and the specific setup of your application. Always refer to the official documentation or the source code for the most accurate and up-to-date information.

If you need to configure options for the HTTP activities, you can do so by passing a lambda to the `AddHttpActivities` method, similar to how it's done in previous versions of Elsa:

```csharp
builder.Services.AddElsa(elsa => {
    // ... other configurations ...

    // Enable HTTP activities with additional configuration.
    elsa.AddHttpActivities(options => {
        // Configure the options here.
    });
    
    // ... other configurations ...
});
```

Make sure to adjust the configuration according to your needs and the structure of your application's configuration files.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
